src/OutputPresenter.ts

```diff
@@ -0,0 +1,268 @@
+    if (this.playedSeconds >= this.currLength.slot[this.playingMedia]) {
+      this.playingMedia += 1;
+      this.playedSeconds = 0;
+    }
+  }
+
+  @mobx.action
+  adjustSound() {
+    if (this.music.currentTime >= this.currLength.end - 4) {
+      if (this.music.volume - 0.2 >= 0) {
+        this.music.volume -= 0.2;
+      }
+    } else if (this.music.currentTime - 4 <= this.currLength.start) {
+      if (this.music.volume + 0.1 <= 1) {
+        this.music.volume += 0.1;
+      }
+    }
+  }
+  @mobx.action
+  incrementPlayingMedia(): void {
+    this.playingMedia++;
+  }
+
+  @mobx.computed
+  get currSlotLength(): number {
+    return this.currLength.slot.length;
+  }
+
+  @mobx.computed
+  get totalVideoDuration(): number {
+    return this.currLength.totalDuration;
+  }
+
+  @mobx.computed
+  get currOverallPlayedSeconds(): number {
+    return this.overallPlayedSeconds;
+  }
+
+  @mobx.action
+  setCurrMood(newTemplate: Mood, filesLength: number) {
+    this.currMood = newTemplate;
+    this.setCurrTrack(0, filesLength);
+  }
+
+  @mobx.computed
+  get templateMood(): string {
+    return this.currMood.style;
+  }
+  @mobx.action
+  setTrackIndex(newIndex: number): void {
+    this.trackIndex = newIndex;
+  }
+  // lets say current is long, user changes mood, and the current moodTemplate do not support long
+  getSuitableLength(filesLength: number): string {
+    if (this.canShowDuration(filesLength, this.lengthIndex)) {
+      return this.lengthIndex;
+    }
+    if (
+      this.lengthIndex === "long" &&
+      this.canShowDuration(filesLength, "medium")
+    ) {
+      return "medium";
+    }
+    return "short";
+  }
+
+  @mobx.action
+  setCurrTrack(trackIndex: number, filesLength: number): void {
+    this.trackIndex = trackIndex;
+    this.currTrack = this.currMood.tracks[trackIndex];
+    this.music.pause();
+    this.music = new Audio(this.currTrack.musicTrack);
+    this.setCurrLength(this.getSuitableLength(filesLength));
+  }
+
+  @mobx.action
+  prevCurrTrack(filesLength: number): void {
+    if (this.trackIndex - 1 < 0) {
+      this.setCurrTrack(this.currMood.tracks.length - 1, filesLength);
+    } else {
+      this.setCurrTrack(this.trackIndex - 1, filesLength);
+    }
+  }
+
+  @mobx.action
+  nextCurrTrack(filesLength: number): void {
+    if (this.trackIndex + 1 >= this.currMood.tracks.length) {
+      this.setCurrTrack(0, filesLength);
+    } else {
+      this.setCurrTrack(this.trackIndex + 1, filesLength);
+    }
+  }
+
+  @mobx.action
+  setCurrLength(newLength: string) {
+    if (
+      newLength === "short" ||
+      newLength === "medium" ||
+      newLength === "long"
+    ) {
+      this.lengthIndex = newLength;
+      this.currLength = this.currTrack[newLength];
+      this.resetVideo();
+    } else {
+      alert(newLength + " is not short, medium or long");
+    }
+  }
+
+  canShowDuration(filesLength: number, duration: string): boolean {
+    if (duration === "short") return true;
```

Comment:
nit: this `duration` should be `length` (duration = number, length = 'short' | 'medium' | 'long', as per your template)